stdlib/public/core/StringStorage.swift

```diff
@@ -390,6 +404,26 @@ extension __StringStorage {
     return __StringStorage.create(
       realCodeUnitCapacity: realCapacity, countAndFlags: countAndFlags)
   }
+  
+  @_effects(releasenone)
+  internal static func create(
+    unsafeUninitializedCapacity capacity: Int,
+    initializingUncheckedUTF8With initializer: (
+      _ buffer: UnsafeMutableBufferPointer<UInt8>,
+      _ initializedCount: inout Int
+    ) throws -> Void
+  ) rethrows -> __StringStorage {
+    let storage = __StringStorage.create(
+      capacity: capacity,
+      countAndFlags: CountAndFlags(mortalCount: 0, isASCII: false)
+    )
+    let buffer = UnsafeMutableBufferPointer(start: storage.mutableStart,
+                                            count: capacity)
+    var count = 0
+    try initializer(buffer, &count)
+    storage._countAndFlags = CountAndFlags(mortalCount: count, isASCII: false)
```

Comment:
Maybe a comment explaining that the caller will handle validation and ASCII-ness for us.

Reply:
Yeah good call